server/raft.go

```diff
@@ -4063,13 +4051,15 @@ const (
 	noVote   = _EMPTY_
 )
 
-func (n *raft) switchToFollower(leader string) {
+func (n *raft) switchToFollower(leader string, locked bool) {
```

Comment:
Let's keep this one as unlocked and add in a switchToFollowerLocked() that is called from switchToFollower().

Reply:
This will also mean adding `stepdown()` and `stepdownLocked()`. If you're OK with that then I can make the change in the morning.

Reply:
Make reading easier, I was struggling til I saw what the bool meant way down in the PR. Meaning when we go back to this code after some time away will have similar issue IMO.